Release master doesn't build: no method named `debug_check_unchanged` found for reference `&DelayedReceiptQueue`

Trying to do a release build on the current master (400bdf19fc3e7a31f5733a160927c2bb72753ceb) fails.

```
$ cargo build -p neard --release
   Compiling neard v0.0.0 (/home/jan/code/nearcore/neard)
   Compiling near-jsonrpc-client v0.0.0 (/home/jan/code/nearcore/chain/jsonrpc/client)
   Compiling near-store v0.0.0 (/home/jan/code/nearcore/core/store)
error[E0599]: no method named `debug_check_unchanged` found for mutable reference `&mut DelayedReceiptQueue` in the current scope
  --> core/store/src/trie/receipts_column_helper.rs:41:18
   |
41 |             self.debug_check_unchanged(state_update);
   |                  ^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut DelayedReceiptQueue`

error[E0599]: no method named `debug_check_unchanged` found for mutable reference `&mut DelayedReceiptQueue` in the current scope
  --> core/store/src/trie/receipts_column_helper.rs:55:18
   |
55 |             self.debug_check_unchanged(state_update);
   |                  ^^^^^^^^^^^^^^^^^^^^^ method not found in `&mut DelayedReceiptQueue`

error[E0599]: no method named `debug_check_unchanged` found for reference `&DelayedReceiptQueue` in the current scope
  --> core/store/src/trie/receipts_column_helper.rs:80:18
   |
80 |             self.debug_check_unchanged(trie);
   |                  ^^^^^^^^^^^^^^^^^^^^^ method not found in `&DelayedReceiptQueue`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `near-store` (lib) due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
```

The error originates from this function, added in https://github.com/near/nearcore/pull/11159:
```rust
/// Check the queue has not been modified in the trie view.
    ///
    /// This is a semi-expensive operation. The values should be cached in
    /// memory in at least one layer. But we still want to avoid it in
    /// production.
    #[cfg(debug_assertions)]
    fn debug_check_unchanged(&self, trie: &dyn TrieAccess) {
        debug_assert_eq!(
            self.indices,
            get(trie, &TrieKey::DelayedReceiptIndices).unwrap().unwrap_or_default()
        );
    }
```

Recently we also had some feature changes, could be related: https://github.com/near/nearcore/pull/11158

/cc @jakmeier @wacban

--- a/core/store/src/trie/receipts_column_helper.rs
+++ b/core/store/src/trie/receipts_column_helper.rs
@@ -37,9 +37,7 @@ impl DelayedReceiptQueue {
         state_update: &mut TrieUpdate,
         receipt: &Receipt,
     ) -> Result<(), StorageError> {
-        if cfg!(debug_assertions) {
-            self.debug_check_unchanged(state_update);
-        }
+        self.debug_check_unchanged(state_update);
         let index = self.indices.next_available_index;
         set(state_update, TrieKey::DelayedReceipt { index }, receipt);
 
@@ -51,9 +49,7 @@ impl DelayedReceiptQueue {
     }
 
     pub fn pop(&mut self, state_update: &mut TrieUpdate) -> Result<Option<Receipt>, StorageError> {
-        if cfg!(debug_assertions) {
-            self.debug_check_unchanged(state_update);
-        }
+        self.debug_check_unchanged(state_update);
         if self.indices.first_index >= self.indices.next_available_index {
             return Ok(None);
         }
@@ -76,9 +72,7 @@ impl DelayedReceiptQueue {
     }
 
     pub fn iter<'a>(&self, trie: &'a dyn TrieAccess) -> ReceiptIterator<'a> {
-        if cfg!(debug_assertions) {
-            self.debug_check_unchanged(trie);
-        }
+        self.debug_check_unchanged(trie);
         ReceiptIterator {
             trie_keys: Box::new(
                 (self.indices.first_index..self.indices.next_available_index)
@@ -99,6 +93,10 @@ impl DelayedReceiptQueue {
             self.indices,
             get(trie, &TrieKey::DelayedReceiptIndices).unwrap().unwrap_or_default()
         );
+    }
+    #[cfg(not(debug_assertions))]
+    fn debug_check_unchanged(&self, _trie: &dyn TrieAccess) {
+        // nop in release build
     }
 }
 

